tests/explore_test.sh

```diff
@@ -5,21 +5,111 @@
 
 function suite
 {
-  suite_addTest "testExplore"
+  suite_addTest "explore_files_under_git_repo_Test"
+  suite_addTest "explore_git_log_Test"
+  suite_addTest "explore_parser_Test"
 }
 
-function testExplore
+function setupExplore()
 {
-  local MSG_OUT="GNU"
-  local LOG_OUT="Initial commit"
+  local -r test_path="tests/.tmp"
+  local -r current_path="$PWD"
 
-  assertEquals "Expected an error message." "Expected path or 'log'" "$(explore)"
+  rm -rf "$test_path"
+
+  mkdir -p "$test_path"
+  cd "$test_path"
+  # Setup git repository for test
+  git init &> /dev/null
+
+  for commit in {1..5}; do
+    local file_name="file_$commit.txt"
+    echo "This is file $commit" > "$file_name"
+    git add "$file_name" &> /dev/null
+    git commit -m "Commit number $commit" &> /dev/null
+  done
+
+  cd "$current_path"
+}
+
+function tearDownExplore()
+{
+  local -r test_path="tests/.tmp"
+
+  rm -rf "$test_path"
+}
+
+function explore_files_under_git_repo_Test
+{
+  local ID
+  local MSG_OUT
+
+  setupExplore
+
+  ID=1
+  MSG_OUT="Expected string or 'log'"
+  output=$(explore)
+  assertEquals "($ID) - Expected an error message." "$MSG_OUT" "$output"
+
+  ID=2
   MSG_OUT="camel_case(void)"
-  assertEquals "$MSG_OUT" "$(explore "camel_case" tests/samples | cut -d ' ' -f2 | sed -n -e 2p)"
-  assertEquals "$LOG_OUT" "$(explore log LICENSE | grep "Initial commit" | awk '{print $1, $2}')"
+  output=$(explore "camel_case" tests/samples | cut -d ' ' -f2 | sed -n -e 2p)
```

Comment:
Since we now have a fake repo created by `setupExplore()` can we use it instead of kw's repo? I think it will make the test more robust.

Reply:
I did not get your idea. The setup function creates a fake project with fake commits, and it does not create any real file for simulate search. Notice that we search for something in a file in this test, I can't see any robustness advantage by making this file isolate. I believe that we are safer by using our samples.

Reply:
What I meant is that this test performs a search in kw's repository and not in the fake repository we created for the tests. Since we already have the fake repo, I suggest changing the above lines for something like:

```
MSG_OUT="file_1.txt:1:This is file 1"
output=$(explore "is file 1" "$test_path/1")
```

The advantage is that we don't have to rely on kw's repo for the test, so if we have to change the `tests/samples` or `LICENSE` (as in the other test bellow) we don't have to adjust the tests as well.

Reply:
I've one idea about this, I'll implement it today and update this PR

Reply:
Done :)